lib/src/logic.dart

```diff
@@ -575,6 +575,74 @@ class Logic {
   /// Greater-than-or-equal-to.
   Logic operator >=(dynamic other) => GreaterThanOrEqual(this, other).out;
 
+  /// Shorthand for a [Conditional] which increments this by [incrVal]
+  ///
+  /// By default for a [Logic] var, if no [incrVal] is provided result is ++var
+  /// else result is var+=[incrVal]
+  ///
+  /// ```dart
+  ///
+  /// // Given a and b Logic input and piOut as output
+  /// Combinational([
+  ///   piOut < a,
+  ///   piOut.incr(b),
+  /// ]);
+  ///
+  /// ```
+  ///
+  ConditionalAssign incr([dynamic incrVal]) =>
+      this < ((incrVal != null) ? this + incrVal : this + Const(1));
```

Comment:
Could use this to make it shorter and more readable
`this < (this + incrVal ?? 1)`